torch/_dynamo/utils.py

```diff
@@ -4474,3 +4474,30 @@ def get_optimize_ddp_mode():
         f"Invalid dynamo config optimize_ddp value {mode=}"
     )
     return mode
+
+
+@contextmanager
+def maybe_disable_inference_mode() -> Generator[None, None, None]:
+    is_inference_mode_on = (
+        config.fake_tensor_disable_inference_mode and torch.is_inference_mode_enabled()
+    )
+    if is_inference_mode_on:
+        with (
+            torch.inference_mode(False),
+            torch.no_grad(),
+            torch._subclasses.meta_utils.disable_inference_mode_for_fake_prop(),
+        ):
+            yield
```

Comment:
To check my understanding, you're using maybe_disable_inference_mode to chain context managers together? (torch.inference_mode(False),
            torch.no_grad(),
         torch._subclasses.meta_utils.disable_inference_mode_for_fake_prop(),)

Reply:
Yes. Earlier this was done in an inlined manner in the `convert_frame.py` file but it was making code difficult to read.